src/resources/subscriptions-resource.js

```diff
@@ -1,92 +1,93 @@
-            const params = {expand};
-            return apiHandler.get(`subscriptions/${id}`, params);
-        },
-
-        create({id = '', data, expand = null}) {
-            const params = {expand};
-            return apiHandler.create(`subscriptions/${id}`, id, data, params);
-        },
-
-        update({id, data, expand = null}) {
-            const params = {expand};
-            return apiHandler.put(`subscriptions/${id}`, data, params);
-        },
-
-        changePlan({id, data}) {
-            return apiHandler.post(`subscriptions/${id}/change-plan`, data);
-        },
-
-        getAllUpcomingInvoices({id=null, limit = null, offset = null, sort = null, filter = null, expand = null} = {}) {
-            const params = {
-                limit,
-                offset,
-                sort,
-                filter,
-                expand,
-            };
-            return apiHandler.getAll(`subscriptions/${id}/upcoming-invoices`, params);
-        },
-
-        issueUpcomingInvoice({id, invoiceId, data}) {
-            return apiHandler.post(`subscriptions/${id}/upcoming-invoices/${invoiceId}/issue`, data);
-        },
-
-        getAllTimelineMessages({id = null, limit = null, offset = null, sort = null, filter = null, q = null} = {}) {
-            const params = {
-                limit,
-                offset,
-                sort,
-                filter,
-                q
-            };
-            return apiHandler.getAll(`subscriptions/${id}/timeline`, params);
-        },
-
-        getTimelineMessage({id, messageId}) {
-            return apiHandler.get(`subscriptions/${id}/timeline/${messageId}`);
-        },
-
-        deleteTimelineMessage({id, messageId}) {
-            return apiHandler.delete(`subscriptions/${id}/timeline/${messageId}`);
-        },
-
-        createTimelineComment({id, data}) {
-            return apiHandler.create(`subscriptions/${id}/timeline`, '', data);
+  return {
+    getAll({
+      filter = null,
+      sort = null,
+      limit = null,
+      offset = null,
+      q = null,
+      expand = null,
+    } = {}) {
+      const params = {filter, sort, limit, offset, q, expand};
+      return apiHandler.getAll(`subscriptions`, params);
+    },
+    create({id = '', data, expand = null}) {
+      const params = {expand};
+      return apiHandler.create(`subscriptions/${id}`, id, data, params);
+    },
+    get({id, expand = null}) {
+      const params = {expand};
+      return apiHandler.get(`subscriptions/${id}`, params);
+    },
+    update({id, data, expand = null}) {
+      const params = {expand};
+      return apiHandler.put(`subscriptions/${id}`, data, params);
+    },
+    changePlan({id, data}) {
+      return apiHandler.post(`subscriptions/${id}/change-plan`, data);
+    },
+    createInterimInvoice({id, data}) {
+      return apiHandler.post(`subscriptions/${id}/interim-invoice`, data);
+    },
+    getAllUpcomingInvoices({id = null, expand = null} = {}) {
```

Comment:
A few arguments missing here, we may need to check the definitions.

Reply:
In this case, the specification looks correct. I checked it with @iamdevice when analyzing the discrepancies. 
We are not expecting `filter, sort, offset, limit` in the backend side.

Reply:
@cesarlevel get all upcoming invoices means we're looking for upcoming invoices for specified subscription/order.